righttyper/righttyper_runtime.py

```diff
@@ -307,45 +307,50 @@ def get_full_type(value: Any, /, use_jaxtyping: bool = False, depth: int = 0) ->
 
     if isinstance(value, dict):
         t = type(value)
-        if value:
-            el = value.random_item() if isinstance(value, RandomDict) else sample_from_collection(value.items())
-            args = tuple(get_full_type(fld, depth=depth+1) for fld in el)
-        else:
-            args = (TypeInfo("typing", "Never"), TypeInfo("typing", "Never"))
+        args = (TypeInfo("typing", "Never"), TypeInfo("typing", "Never"))
+        try:
+            if value:
+                el = value.random_item() if isinstance(value, RandomDict) else sample_from_collection(value.items())
+                args = tuple(get_full_type(fld, depth=depth+1) for fld in el)
+        except Exception: pass
         module = "" if t.__module__ == "builtins" else t.__module__
         return TypeInfo(module, t.__qualname__, args=args)
     elif isinstance(value, (list, set)):
         t = type(value)
-        if value:
-            el = sample_from_collection(value)
-            args = (get_full_type(el, depth=depth+1),)
-        else:
-            args = (TypeInfo("typing", "Never"),)
+        args = (TypeInfo("typing", "Never"),)
+        try:
+            if value:
+                el = sample_from_collection(value)
+                args = (get_full_type(el, depth=depth+1),)
+        except Exception: pass
         module = "" if t.__module__ == "builtins" else t.__module__
         return TypeInfo(module, t.__qualname__, args=args)
     elif (t := _is_instance(value, (abc.KeysView, abc.ValuesView))):
-        if value:
-            el = sample_from_collection(value)
-            args = (get_full_type(el, depth=depth+1),)
-        else:
-            args = (TypeInfo("typing", "Never"),)
+        args = (TypeInfo("typing", "Never"),)
+        try:
+            if value:
+                el = sample_from_collection(value)
+                args = (get_full_type(el, depth=depth+1),)
+        except Exception: pass
         return TypeInfo("typing", t.__qualname__, args=args)
     elif isinstance(value, abc.ItemsView):
-        if value:
-            el = sample_from_collection(value)
-            args = tuple(get_full_type(fld, depth=depth+1) for fld in el)
-        else:
-            args = (TypeInfo("typing", "Never"), TypeInfo("typing", "Never"))
+        args = (TypeInfo("typing", "Never"), TypeInfo("typing", "Never"))
+        try:
+            if value:
+                el = sample_from_collection(value)
+                args = tuple(get_full_type(fld, depth=depth+1) for fld in el)
+        except Exception: pass
         return TypeInfo("typing", "ItemsView", args=args)
     elif isinstance(value, tuple):
         if isinstance_namedtuple(value):
             t = type(value)
             return TypeInfo(t.__module__, t.__qualname__)
         else:
-            if value:
-                args = tuple(get_full_type(fld, depth=depth+1) for fld in value)
-            else:
-                args = tuple()
+            args = tuple()
```

Comment:
to just type with `tuple` (without arguments), simply set `args = None` here.  I am not sure why `args = tuple()` before

Reply:
Ugh, actually, I was wrong. Leave it at `tuple()`, please :/